ssz/src/main/java/tech/pegasys/teku/ssz/backing/view/AbstractImmutableContainer.java

```diff
@@ -28,16 +28,17 @@
 /** Handy base class for immutable containers */
 public abstract class AbstractImmutableContainer extends ContainerViewReadImpl {
 
-  public AbstractImmutableContainer(ContainerViewType<? extends AbstractImmutableContainer> type) {
+  protected AbstractImmutableContainer(
+      ContainerViewType<? extends AbstractImmutableContainer> type) {
```

Comment:
Is it something which is enforced now?

Reply:
Yes. It's technically right that there's no point having a `public` constructor for an abstract class but I'm not sure I care. If it gets annoying I'd be happy to disable it.  I just generally try to minimise the number of rules we disable unless there's a real reason to.